### PR TITLE
Bigtable: create snippets.py for python docs.

### PR DESCRIPTION
**Pull request by sangramql**

### CODE REVIEW

File: bigtable/nox.py [hunk 1]
```diff
@@ -98,27 +98,55 @@ def system(session, py):
...
+    session.run('py.test', os.path.join('../docs/bigtable', \
+                                        'snippets.py'), *session.posargs)
+
+
+
 @nox.session
 def lint(session):
     """Run linters.
-
+
```
[the remark this answers is not in the document]
Comment (sangramql) [reply bigtable/nox.py]: Done

File: docs/bigtable/snippets.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+def _name_and_doc(func):
+    return func.__name__, func.__doc__
+
+
+def main():
+    client = bigtable.Client(project='my-project', admin=True)
+    for example in _find_examples():
+        to_delete = []
+        print '%-25s: %s' % _name_and_doc(example)
```
[the remark this answers is not in the document]
Comment (sangramql) [reply docs/bigtable/snippets.py]: Done

**Comment by sduskis** [comment]
@sumit-ql, can you please update bigtable/nox.ph?

**Comment by tseaver** [comment]
@sduskis 

> @sumit-ql, can you please update bigtable/nox.py?

After #6175, the new location is `bigtable/noxfile.py`.

**Comment by tseaver** [comment]
@sangramql The file `nox.py` has been replaced with `noxfile.py`.  To resolve conflicts, please merge the current `master` into your branch, and re-apply the `snippets` session there.

You will need to update Nox via:

```bash
$ /path/to/venv/bin/pip uninstall nox-automation
$ /path/to/venv/bin/pip install nox
```

**Comment by sangramql** [comment]
Hi @tseaver , Fixed the tests, could you please review it.

**Comment by sduskis** [comment]
@sumit-ql, it looks like there's a CircleCI failure:

> sphinx.errors.SphinxWarning: /var/code/gcp/docs/index.rst:1:toctree contains reference to nonexisting document 'bigtable/index'
>
> Warning, treated as error:
/var/code/gcp/docs/index.rst:1:toctree contains reference to nonexisting document 'bigtable/index'

**Comment by sangramql** [comment]
Hi @tseaver , Fixed the tests, could you please review it.

**Comment by sangramql** [comment]
@tseaver can you suggest the help on this failure. index link is failing for some files, but looks like passing for some. Do we need to change all the links?

**Comment by tseaver** [comment]
@sangramql Since PR #6014, `docs/bigtable` has been a symlink to `bigtable/docs`, which does include `index.rst`.  Maybe you need to rebase your `create_snippet` branch on the current `master`?

File: bigtable/docs/snippets.py [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+def test_bigtable_list_clusters():
+    # [START bigtable_list_clusters]
+    from google.cloud.bigtable import Client
+
+    client = Client(admin=True)
+    instance = client.instance(INSTANCE_ID)
+    (clusters_list, failed_locations_list) = instance.list_clusters()
+    # [END bigtable_list_clusters]
+
```
Comment (AVaksman) [inline bigtable/docs/snippets.py]: Please add `list_clusters_in_project` here
```python
def test_bigtable_list_clusters_in_project():
    # [START test_bigtable_list_clusters_in_project]
    from google.cloud.bigtable import Client
    client = Client(admin=True)
    (clusters_list, failed_locations_list) = client.list_clusters()
    # [END test_bigtable_list_clusters_in_project]
    assert clusters_list.__len__() is not 0
```

File: bigtable/docs/snippets.py [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    client = Client(admin=True)
+    (instances_list, failed_locations_list) = client.list_instances()
+    # [END bigtable_list_instances]
+
+    assert instances_list.__len__() is not 0
+
+
+def test_bigtable_list_clusters():
+    # [START bigtable_list_clusters]
```
Comment (AVaksman) [inline bigtable/docs/snippets.py]:
```suggestion
    # [START bigtable_list_clusters_on_instance]
```

File: bigtable/docs/snippets.py [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    from google.cloud.bigtable import Client
+
+    client = Client(admin=True)
+    instance = client.instance(INSTANCE_ID)
+    (clusters_list, failed_locations_list) = instance.list_clusters()
+    # [END bigtable_list_clusters]
```
Comment (AVaksman) [inline bigtable/docs/snippets.py]:
```suggestion
    # [END bigtable_list_clusters_on_instance]
```

File: bigtable/google/cloud/bigtable/client.py [hunk 6]
```diff
@@ -256,6 +268,12 @@ def list_instances(self):
     def list_clusters(self):
         """List the clusters in the project.
 
+        For example:
+
+        .. literalinclude:: snippets.py
+            :start-after: [START bigtable_list_clusters]
```
Comment (AVaksman) [inline bigtable/google/cloud/bigtable/client.py]: Please change to
```python
            :start-after: [START bigtable_list_clusters_in_project]
```
after adding that snippet

File: bigtable/google/cloud/bigtable/client.py [hunk 7]
```diff
@@ -256,6 +268,12 @@ def list_instances(self):
...
+            :end-before: [END bigtable_list_clusters]
```
Comment (AVaksman) [inline bigtable/google/cloud/bigtable/client.py]: Please change to 
```python
            :end-before: [END bigtable_list_clusters_in_project]
```
after adding that snippet.

File: bigtable/google/cloud/bigtable/instance.py [hunk 8]
```diff
@@ -385,6 +409,12 @@ def cluster(self, cluster_id, location_id=None,
     def list_clusters(self):
         """List the clusters in this instance.
 
+        For example:
+
+        .. literalinclude:: snippets.py
+            :start-after: [START bigtable_list_clusters]
```
Comment (AVaksman) [inline bigtable/google/cloud/bigtable/instance.py]:
```suggestion
            :start-after: [START bigtable_list_clusters_on_instance]
```

File: bigtable/google/cloud/bigtable/instance.py [hunk 9]
```diff
@@ -385,6 +409,12 @@ def cluster(self, cluster_id, location_id=None,
...
+            :end-before: [END bigtable_list_clusters]
```
Comment (AVaksman) [inline bigtable/google/cloud/bigtable/instance.py]:
```suggestion
            :end-before: [END bigtable_list_clusters_on_instance]
```

**Comment by AVaksman** [comment]
@sangramql  Please also add snippets for
* `app_profile`
* `list_app_profiles`
* `get_iam_policy`
* `set_iam_policy`
* `test_iam_permissions`
to complete examplese for `client'.py`, `cluster.py` and `instance.py`

**Comment by tseaver** [comment]
Unrelated Bigtable lint error, fixed in #6309.

Similar unrelated Firestore lint error (why is it even running?), which will be fixed via #6291.

**Comment by tseaver** [comment]
@sangramql Thanks for sticking with this PR!  If you can address the suggestions / requested new snippets from @AVaksman, I think we'll be ready to merge.

**Comment by sangramql** [comment]
@tseaver, @AVaksman ,
I have fixed the review comments, requesting to please review.
I was not able to add get_iam_policy and set_iam_policy as those test failure due to insufficient  permissions. If needed I can add to snippets in separate PR, as those already have code in documentations as of now.

**Comment by tseaver** [comment]
@sangramql I'm pretty sure that if you can create an instance, you should be able to call `get_iam_policy` / `set_iam_policy` / `test_iam_permissions` on it.

The only other missing snippet that I see requested by @AVaksman is for `instance.update.

**Comment by sangramql** [comment]
@tseaver , @AVaksman Please review the updated code with all the reviews implemented.

**Comment by tseaver** [comment]
@sangramql we still need snippets for `instance.update` and `instance.test_iam_permissions`.

**Comment by sangramql** [comment]
@tseaver , @AVaksman added the snippet for get and set iam policy.
Please review.

**Comment by sangramql** [comment]
> @sangramql we still need snippets for `instance.update` and `instance.test_iam_permissions`.

Added snippets for both the methods. [update](https://github.com/googleapis/google-cloud-python/pull/6012/files#diff-84fff32d9b7e8b90fe48983de3c0c847R258-R268) and [iam_permissions](https://github.com/googleapis/google-cloud-python/pull/6012/files#diff-84fff32d9b7e8b90fe48983de3c0c847R360-R370).

**Pull request merged**